src/Objects/Containers/LinkedCell/BoundaryHandler.cpp

```diff
@@ -0,0 +1,132 @@
+//
+// Created by Stefanie Blattenberger on 27/11/2024.
+//
+
+#include "BoundaryHandler.h"
+#include <math.h>
+#include <exception>
+#include <iostream>
+#include <algorithm>
+#include "spdlog/spdlog.h"
+#include "utils/ArrayUtils.h"
+
+#include "../../../Calculator/Calculator.h"
+#include "../../../Calculator/LennardJonesCalculator.h"
+
+Calculators::LennardJonesCalculator calculator = Calculators::LennardJonesCalculator();
+
+BoundaryHandler::BoundaryHandler(double s, std::array<bool, 6> t, ParticleContainers::LinkedCellContainer& container) :
+sigma {s}, type {t}, container {container}, minDist {std::pow(2.0, 1.0/6.0) * sigma}, 
+boundaries {{0, container.getDomainSize()[0], container.getDomainSize()[1], 0, container.getDomainSize()[2], 0}} {
+//initializeBoundaries();
+//SPDLOG_INFO("type set to {} {} {} {}", type[0], type[1], type [2], type [3]);
+
+};
+
+
+void BoundaryHandler::handleBoundaries(){
+            handleReflecting();
+            handleOutflow();
+    }
+
+void BoundaryHandler::handleOutflow(){
+    //SPDLOG_INFO("handle outflow");
+    for (auto cell : container.getHaloCells()) {
+        for (auto p : cell.get().getParticlesInCell()) {
+            for (int i = 0; i < 4; i++) { //FOR 3D: 6
+
+                if (type[i] == 1) {
+                    continue;
+                }
+
+                double dist = calculateDistance(*p, i);
+
+                if (dist <= container.getCellSizePerDimension()[i/2]) { //this is a cell at the border that we are looking at
```

Comment:
Good practice: I would add `static_cast<int>(i/2)` to be safe from unwanted changes in future